pkg/hostagent/requirements.go

```diff
@@ -77,7 +81,14 @@ func prefixExportParam(script string) (string, error) {
 
 	// TODO we should have a symbolic constant for `/mnt/lima-cidata`
 	exportParam := `while read -r line; do [ -n "$line" ] && export "$line"; done<<EOF\n$(sudo cat /mnt/lima-cidata/param.env)\nEOF\n`
-	return fmt.Sprintf("#!/bin/bash -c $'%s%s'\n%s", exportParam, interpreter, script), nil
+
+	// double up all '%' characters so we can pass them through unchanged in the format string of printf
+	interpreter = strings.ReplaceAll(interpreter, "%", "%%")
+	exportParam = strings.ReplaceAll(exportParam, "%", "%%")
+	// strings will be interpolated into single-quoted strings, so protect any existing single quotes
+	interpreter = strings.ReplaceAll(interpreter, "'", `'"'"'`)
+	exportParam = strings.ReplaceAll(exportParam, "'", `'"'"'`)
```

Comment:
All these replacements are not really needed because the input strings don't include `%` or `'` characters. but it feels like the right thing to do to make them correct for that case as well.